mypy/checker.py

```diff
@@ -1827,13 +1827,15 @@ def visit_try_stmt(self, s: TryStmt) -> Type:
     def visit_except_handler_test(self, n: Node) -> Type:
         """Type check an exception handler test clause."""
         type = self.accept(n)
-        if isinstance(type, AnyType):
-            return type
 
         all_types = []  # type: List[Type]
         test_types = type.items if isinstance(type, TupleType) else [type]
 
         for ttype in test_types:
+            if isinstance(ttype, AnyType):
+                all_types.append(ttype)
```

Comment:
After looking at your tests, I take it back.  The way you're doing it here correctly catches additional non-Exception-derived types that come after an `Any`.